source/NonInvasiveKeyboardHook/NonInvasiveKeyboardHookLibrary/KeyboardHookManager.cs

```diff
@@ -217,36 +225,28 @@ private static IntPtr SetHook(LowLevelKeyboardProc proc)
         {
             var userLibrary = LoadLibrary("User32");
 
-            return SetWindowsHookEx(WH_KEYBOARD_LL, proc,
-                userLibrary, 0);
+            return SetWindowsHookEx(WH_KEYBOARD_LL, proc, userLibrary, 0);
         }
-        
+
         private IntPtr HookCallback(int nCode, IntPtr wParam, IntPtr lParam)
         {
             if (nCode >= 0)
             {
                 var vkCode = Marshal.ReadInt32(lParam);
 
-                // Debug.WriteLine("Starting");
-                // To prevent slowing keyboard input down, we use handle keyboard inputs in a separate thread
-                ThreadPool.QueueUserWorkItem(this.HandleSingleKeyboardInput, new KeyboardParams(wParam, vkCode));
```

Comment:
Although it has been a long time since I have coded/used this package, I recall that this had an important positive impact on performance.

Unfortunately, the new behavior you are trying to introduce requires (at the very least) to check if the pressed keys are assigned to a blocking hotkey before proceeding. I do believe that looking up the hotkey is fast enough for us to move it to the main thread, but as soon as we find out that a hotkey is non-blocking I want it to be handled by the thread pool like it used to be.

The bottom line is we have two choices:
- either move the parameter up to the KeyboardManager's constructor parameters and then maintain the old thread pool behavior when non-blocking,
- or make the handling of non-blocking hotkeys run in a new thread to maintain the old behavior